### PR TITLE
[MacSDK] Keep logged post-install script output from previous installs

### PR DESCRIPTION
**Pull request by alexischr**
This is helpful for issues such as https://devdiv.visualstudio.com/DevDiv/_workitems/edit/596552 OC, where the user is asked to both provide these logs for debugging, and also reinstall the SDK to resolve the issue

### CODE REVIEW

**Comment by alexischr** [comment]
@monojenkins build pkg

**Comment by akoeplinger** [comment]
Actually this won't work since we're cleaning the mono install directory in the preinstall: https://github.com/mono/mono/blob/macsdk-append-logs/packaging/MacSDK/packaging/resources/preinstall#L3-L9

Maybe we could put it in the temp folder? that'd at least survive until a restart.

**Comment by alexischr** [comment]
@akoeplinger I had missed the addition of the preinstall script! Do you think it's reasonable to modify that script so that it rm's "the catalog of files in the mono prefix minus any postinstall*.log" files or something to that effect?

**Comment by akoeplinger** [comment]
It looks like it's not easy to exclude files/folders from `rm`.

I think what we should do instead is not log into the Frameworks directory but into `/Library/Logs`.

**Comment by alexischr** [comment]
@monojenkins build pkg

**Comment by alexischr** [comment]
```
checking for a Python interpreter with version >= 3.2... none
configure: error: no suitable Python interpreter found
```
Looks like we need to add Python 3 to our macOS package-generating bots.

**Comment by lambdageek** [comment]
@monojenkins build pkg

File: packaging/MacSDK/packaging/resources/postinstall [hunk 1]
```diff
@@ -60,16 +60,23 @@ if [ -d ${FW_CURRENT} ]; then
     cd ${FW_CURRENT}/etc
     # Make sure we run the files we lay down, and not other stuff installed on the system
     export PATH="${FW_CURRENT}/bin:$PATH"
+    LOG_DIR=/Library/Logs/Mono.framework
```
Comment (akoeplinger) [inline packaging/MacSDK/packaging/resources/postinstall]: pretty sure you need to `mkdir` the folder before you can use it.

Comment (alexischr) [reply packaging/MacSDK/packaging/resources/postinstall]: 🤦‍♂

**Comment by alexischr** [comment]
@monojenkins build pkg

**Comment by akoeplinger** [comment]
I tested it and it seems to work. My postinstall-gdk-pixbuf-query-loaders.log contains this entry:

```
======= Fri Feb 7 11:23:08 CET 2020 =====
g_module_open() failed for /Library/Frameworks/Mono.framework/Versions/6.13.0/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so: dlopen(/Library/Frameworks/Mono.framework/Versions/6.13.0/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so, 10): Symbol not found: _OBJC_CLASS_$_NSFont
  Referenced from: /Library/Frameworks/Mono.framework/Versions/6.13.0/lib/libpangocairo-1.0.0.dylib
  Expected in: flat namespace
 in /Library/Frameworks/Mono.framework/Versions/6.13.0/lib/libpangocairo-1.0.0.dylib
```

I can find a few mentions of this on GitHub (e.g. https://github.com/Automattic/node-canvas/issues/928) but all point back to mono :D

**Comment by alexischr** [comment]
This appears to work. @akoeplinger Yeah we have had the error message for a while, it shows up for everyone it seems, but we haven't found the cause (or a symptom)

**Pull request merged**